lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/MethodProcessor.java

```diff
@@ -91,6 +94,19 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
           "org.enso.interpreter.runtime.state.Stateful",
           "org.enso.interpreter.runtime.type.TypesGen");
 
+  private void registerBuiltinMethod(MethodDefinition methodDefinition, Element element) throws IOException {
+      String tpe = methodDefinition.getType().toLowerCase();
+      if (tpe.isEmpty()) {
+        throw new InternalError("Type of the BuiltinMethod cannot be empty");
+      }
+      FileObject res = processingEnv.getFiler().createResource(
+              StandardLocation.CLASS_OUTPUT, "",
+              MethodDefinition.META_PATH + "/" + tpe + "/" + methodDefinition.getClassName() + ".builtin", element
+      );
```

Comment:
It would be better to collect all these registrations and dump them when the annotation processing is over into a single file. For example the [AbstractServiceProviderProcessor](https://github.com/apache/netbeans/blob/8fd5ef18f1ec27d8669e382c39633001925d5e80/platform/openide.util.lookup/src/org/openide/util/lookup/implspi/AbstractServiceProviderProcessor.java#L97) does that. It checks for `processingOver()` and only then write the whole file(s) down.

Reply:
Thanks! Wasn't aware of this trick. I thought Filer only restricted to either get a (read-only!) resource or create a new resource. I will take a look.

Reply:
Thanks, done